apm-agent-plugins/apm-mongoclient-plugin/src/test/java/co/elastic/apm/agent/mongoclient/MongoClientSyncVersionIT.java

```diff
@@ -0,0 +1,50 @@
+package co.elastic.apm.agent.mongoclient;
+
+import co.elastic.apm.agent.TestClassWithDependencyRunner;
+import org.junit.After;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.util.Arrays;
+
+@RunWith(Parameterized.class)
+public class MongoClientSyncVersionIT {
+
+    private final TestClassWithDependencyRunner runner;
+
+    public MongoClientSyncVersionIT(String version) throws Exception {
+        runner = new TestClassWithDependencyRunner("org.mongodb", "mongo-java-driver", version,
```

Comment:
Amazing!
If we can use the same container instance for all tests (instead of starting and stopping for each test)- that may improve efficiency.
In addition, since this is so easy to do now, there are LOTs of versions tested. It's better to select only the few that represent API changes, unless this is super fast.

Reply:
I thought of that but it would not be that easy to do, especially because we want the `MongoClientSyncInstrumentationIT` to be executable as a standalone test as well.
But even with the container restarting for each test, it's still quite fast.